### DISCUSSION TITLE
Ways to avoid reflections

### DISCUSSION THREAD

**Discussion by plimmplamm** [jomjol/AI-on-the-edge-device#1949] [Ideas]
Hi eveyone. I have tested a lot of diffusers/contraptions to get the onboard LED to shine light evenly onto the digits.
So far a piece of glue stick worked best. Simply cut a 3 cm piece and then stick it with onto the LED (the circular, cut surface).

Works like a charm.

**Comment by hilsonp** [comment]
What do you call a "piece of glue-stick" ?

**Comment by plimmplamm** [comment]
Hot glue gun sticks. I used 3 cm of it. Heat the end up and stick it to the
LED.



-- 
Klaus Crean

**Reply by hilsonp** [reply]
Got it ! Thank you 🙏

**Reply by twodrops** [reply]
I was stupid enough to think that, with glue stick this was meant 😁 Luckily I didn't try it yet. Thanks for clarifying. 
![IMG_20230607_214026](https://github.com/jomjol/AI-on-the-edge-device/assets/103930/bafc9578-d278-4cbd-8bb0-e9c5ea15af31)

**Reply by twodrops** [reply]
Instead I used some translucent ones from these door/furniture dampeners. 
![Screenshot_2023-06-07-21-47-07-21_b5f6883d2c20a96c53babc0b4ac88108](https://github.com/jomjol/AI-on-the-edge-device/assets/103930/353cafcd-907d-47d2-a6aa-71112e68d18a)

Works as well, and removes the pointed flash. I wouldn't say it fully diffuses the flash, but far better than leaving nothing on.

**Comment by forrestfunk** [comment]
Hey, I'm interested in  can you show us a photo how this looks like in your setup? Thanks

**Comment by schneich** [comment]
@plimmplamm,

Did you create something like a 'light-strip' out of glue? I have to test this!

Edit: did anyone try something like this?
https://a.aliexpress.com/_mquMrYu

**Reply by skupu** [reply]
I don't have any problems with reflections - my led settings are 4-7 and I'm using this kind of pipe as holder.
 
![image](https://github.com/jomjol/AI-on-the-edge-device/assets/10194143/ecb60580-2154-416e-9f21-2e3ece3d3f39)

**Reply by twodrops** [reply]
hmmm, I feel it is not really the holder that decides if there is a reflection or not (see pt.3 below) but,

1. The surface of the top covering of the meter (glass, plastic etc.),
2. The material/color of the meter under the covering,
3. Position of flash, not much the height as you cannot control it as you would fix it for a better picture coverage and focus, but the position that can be controlled by turning the ESP32 around. With this, as already mentioned in the documentation, you could control that the flash lands in a part where no data needs to be read
4. The intensity of flash (0 .. 100)
5. The inside color of the holder

This is how mine looks with the 3M dampner above:

![image](https://github.com/jomjol/AI-on-the-edge-device/assets/103930/df72b310-57e5-47e5-af51-57686e4b80a7)

And my holder with black color inside,

![2023-06-18_09h21_48](https://github.com/jomjol/AI-on-the-edge-device/assets/103930/76a75b23-9a7f-4c67-bdda-8c7475780007)

**Reply by skupu** [reply]
Yes, you are right, but the white inside wasn't good enough. And the direction of the flash can be set in my case, as the esp is not fixed but firmly pushed into the pipe end cap.

see here: https://github.com/jomjol/AI-on-the-edge-device/discussions/2471#discussioncomment-6206589

**Comment by schneich** [comment]
It has been a couple of busy months since I was last able to work on this project. I was facing some misinterpretation of some figures (https://github.com/jomjol/AI-on-the-edge-device/discussions/2423) due to heavy reflection. I tried your solution with the clue stick, which worked ok'ish. Then, the misreadings started again.

To get rid of the reflection, I tried this: https://de.aliexpress.com/item/1005003186394069.html

![grafik](https://github.com/jomjol/AI-on-the-edge-device/assets/84641760/a863108c-1662-4575-b331-a758b53066eb)

![grafik](https://github.com/jomjol/AI-on-the-edge-device/assets/84641760/ee0a1c2a-7127-4d6c-8572-c3212de943bc)


![grafik](https://github.com/jomjol/AI-on-the-edge-device/assets/84641760/e2d534f5-243b-42f0-ac75-73f99084a86d)



This seems to help to reduce the reflection. Unfortunately, I still got the misreadings. I probably will have to provide some training data... :-(

**Reply by schneich** [reply]
Update:
In my last post, there are still some reflections on the numbers. The AI struggles to read those figures.
So, I have spend a couple of ours to try different variations:
1. keep the light source close to or above the camera
2. put the light source as close to the glas as possible
3. put the light source somewhere in-between
4. try high and low LED intensity
5. try to put an anti-glare foil on the glas

Key learnings:
- _Where there is light, there will be reflections_.
- As this is an analog water meter, the little wheels are round. Wherever the light source is, there will be reflections on the wheels itself. (shiny plastic...)
- As the glas is ~5mm above the numbers, there will be reflections on the glas. (Anti-glare foil does not work, as it is too far away from the numbers...)
- indirect light would be best, but I was not keen on re-designing my casing again.


My solution:
I was aiming for a good compromise between as little reflections as possible and a good enough contrast.
I have put the light source as close to the glas as possible and as far away as possible from the numbers. -> small angle to reduce the reflections.
As I am using a fibre optic cable to distribute the light, I had to use a lot of tape to cover those areas that cause reflections in the wrong spots.
It is running now for 18h and no misreadings any more. :-)


Some pictures:
![grafik](https://github.com/jomjol/AI-on-the-edge-device/assets/84641760/d9b65384-f172-4d07-b950-c6e96c4945c5)

![grafik](https://github.com/jomjol/AI-on-the-edge-device/assets/84641760/32fe97b3-1e6b-4445-914a-ff2b3b1822e5)

**Comment by sohnvonoff** [comment]
Hallo,
habe gute Erfahrung mit Polarisationsfilterfolie gemacht. jeweils ein kleines Stück vor Objektiv und um 90° verdreht vor LED geklebt. Damit sieht die Kamera nur das Streulicht, die direkten Reflexionen werden unterdrückt.

**Reply by schneich** [reply]
Good idea. Where do you buy the polarisation foile?

**Reply by friedpa** [reply]
Some question, where to buy? This is a really good Idea!

**Comment by sohnvonoff** [comment]
Das hatte ich dafür bestellt: https://www.amazon.de/gp/product/B07KS5B1LH/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1

**Comment by sohnvonoff** [comment]
Aus der Zeit der 3D-Kinos gibt es die 3D-Brillen. Da ist auch Polarisationsfilterfolie in den Gläsern.

**Comment by macxsm** [comment]
I didn't try polarization foil but I had - by far - **best results with plastic packaging foam**. 
Hot glue was somehow OK, but not good - only for mounting the foam :-)
![image](https://github.com/jomjol/AI-on-the-edge-device/assets/17131803/dbcb9c4c-48df-43ad-81e6-619f12128b37)

**Reply by whitedavidp** [reply]
Thanks. I will try this packing foam. In fact, the very piece that my ESP32 camera came stuck into for shipping. Best!

**Reply by vrilcode** [reply]
Didn't work at all for me, not even 4 layers of foam. I guess it's the hot glue doing the effect.

**Comment by Thyraz** [comment]
I also had a lot problems with reflections on my meter with dark round digits.
Polarisation foil was also no solution, as there was too much diffraction in the glass and reflecting elements.
It removed the reflections on the top plate quite good, but not on the digits itself.

I ended up with a 3d printed shroud to bring the light source closer to the surface without external lights or fibre optic cables.
My outer casing is matte white pla, so it works good as a reflector for diffuse lighting.
The inner shroud (that keeps direct lighting away from the scale I wanna read) is printed in matte black pla, to avoid reflections.

That way the light can only pass below the shrouds lip and won't come from an angle that reflects back to the camera.

The downside of this solution is, that it only works for meters with exclusively digital figures.
As the lower part of the meter is hidden behind the shroud and also over-exposed.

![1](https://github.com/user-attachments/assets/03da607d-304b-4c77-8980-c730a9a64859)
![2](https://github.com/user-attachments/assets/818ae6e0-33e3-40ad-9e6b-4f742a5b1c40)


![IMG_4453](https://github.com/user-attachments/assets/8a5075f9-9868-43ad-8c51-ee99c114b4b9)
![IMG_4448](https://github.com/user-attachments/assets/85c3a52c-e65b-4bdb-9672-794890021f77)
![IMG_4437](https://github.com/user-attachments/assets/90c5f156-b852-4ae3-942c-b6e6d506f6a3)
![IMG_4443](https://github.com/user-attachments/assets/4810a2d6-5705-402d-83a4-28c5d3c748c4)

![3](https://github.com/user-attachments/assets/bf45c691-6fc5-4b42-96ae-663a3e1305e9)

https://www.printables.com/model/1564006-ai-on-the-edge-watermeter-housing-for-zehner-count

**Reply by calma76** [reply]
Your solution works perfectly for me. I didn't have a 3D printer, so I recreated your solution using paper and tesa tape. You can see the result below.
![IMG_4909](https://github.com/user-attachments/assets/5660bb3a-5780-4893-8ebc-473f70edb4ec)
![Screenshot 2026-01-11 181454](https://github.com/user-attachments/assets/62102701-2bcb-4a21-a410-487da5f5de61)
![IMG_4907](https://github.com/user-attachments/assets/2e028f3c-1007-426e-8506-cadf280a8ff3)

**Reply by Thyraz** [reply]
That’s a very creative solution with the paper. 😄👍

**Comment by 35C0lDFuSi0n** [comment]
I went for the hot glue stick diffuser and it works. Thanks for the easy cheap trick.

**Comment by bouwew** [comment]
I also went with the hot glue stick. I heated up the end of the glue stick and then pressed it down on a piece of paper.
This resulted in a nice wide flat-shaped mushroom-like head. When the glue had hardened I cut the paper to size and colored it black with a black-permanent pen/marker. Then I cut off a length of 1.5 - 2 cm from the glue stick and placed it in front of the LED.

This wider top with black color blocks the reflection even better.

Also I covered the inside of the pipe with white paper which improves the lighting of the meter.

**Reply by Niklas1E** [reply]
Do you have a picture?

**Comment by jaeti01** [comment]
@Thyraz  - I like your solution. Did you share your 3D printer files somewhere?  I would like to give it a try! Thank you

**Reply by Thyraz** [reply]
Well, your question reminded me once again to finally upload it on Printables. ;)

So here's the link:
https://www.printables.com/model/1564006-ai-on-the-edge-watermeter-housing-for-zehner-count
